pyiron_base/database/generic.py

```diff
@@ -372,6 +373,19 @@ def get_items_sql(self, where_condition=None, sql_statement=None):
             output_list += [dict(zip(col.keys(), tmp_values))]
         return output_list
 
+    def _check_chem_formula_length(self, par_dict):
+        """
+        performs a check whether the length of chemical formula exceeds the defined limit
+        args:
+        par_dict(dict): dictionary of the parameter
+        limit(int): the limit for the length of checmical formular
+        """
+        for key, value in par_dict.items():
```

Comment:
This is very inefficient, it would be better to just check if the chemicalformular key exists.